pkgs/firehose/lib/src/repo.dart

```diff
@@ -55,16 +55,18 @@ class Repository {
     if (pubspecFile.existsSync()) {
       var pubspec = yaml.loadYaml(pubspecFile.readAsStringSync()) as Map;
       var publishTo = pubspec['publish_to'] as String?;
-      if (publishTo != 'none') {
+      if (publishTo != 'none' && !pubspec.containsKey('workspace')) {
         packages.add(Package(directory, this));
+        // There is an assumption here that published, non-workspace packages do
+        // not contain nested published packages.
+        return;
```

Comment:
Note that this could be removed technically.... but I think it probably is common for packages to have nested packages which are test fixtures, and which don't specify `publish_to: none` today, even though they ideally should. Not recursing below any published packages helps to avoid finding those.

Reply:
I think this is reasonable. I don't think we should support arbitrary nesting for packages but just reasonable patterns. That would include things like a relatively flat list of packages in a `pkgs` directory, with an optional parent 'workspace' pubspec, and non-publishable child example packages.